clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java

```diff
@@ -117,10 +119,17 @@ private List<String> readVariablesAsTypeParameters(final MethodInfo methodInfo)
     parameters.forEach(
         pi ->
             ReflectionUtils.doWithFields(
-                pi.getParameterInfo().getType(), f -> result.add(f.getName())));
+                pi.getParameterInfo().getType(), f -> result.add(extractFieldName(f))));
     return result;
   }
 
+  private String extractFieldName(final Field field) {
+    if (field.isAnnotationPresent(JsonProperty.class)) {
+      return field.getAnnotation(JsonProperty.class).value();
```

Comment:
❌ Actually, one final observation. The annotation could be present but the `value` could still be empty. resulting in a wrong field name.

We should make this check and add an extra test with `@JsonProperty` without a value